spring-cloud-starter-stream-processor-image-recognition/src/test/java/org/springframework/cloud/stream/app/image/recognition/processor/inception/ImageRecognitionTensorflowProcessorIntegrationTests.java

```diff
@@ -0,0 +1,144 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.image.recognition.processor.inception;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.stream.app.image.recognition.processor.ImageRecognitionProcessorConfiguration;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.tuple.Tuple;
+import org.springframework.util.StreamUtils;
+
+/**
+ * @author Christian Tzolov
+ */
+@SuppressWarnings("SpringJavaAutowiringInspection")
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+		webEnvironment = SpringBootTest.WebEnvironment.NONE,
+		properties = {
+				"tensorflow.model=http://dl.bintray.com/big-data/generic/tensorflow_inception_graph.pb",
+				"tensorflow.modelFetch=output",
+				"tensorflow.image.recognition.labels=http://dl.bintray.com/big-data/generic/imagenet_comp_graph_label_strings.txt"
+		})
+@DirtiesContext
+public abstract class ImageRecognitionTensorflowProcessorIntegrationTests {
+
+	@Autowired
+	protected Processor channels;
+
+	@Autowired
+	protected MessageCollector messageCollector;
+
+
+	@TestPropertySource(properties = {
+			"tensorflow.mode=header"
+	})
+	public static class OutputInHeaderTests extends ImageRecognitionTensorflowProcessorIntegrationTests {
+
+		@Test
+		public void testEvaluationPositive() throws IOException {
+			try (InputStream is = new ClassPathResource("/images/panda.jpeg").getInputStream()) {
+
+				byte[] image = StreamUtils.copyToByteArray(is);
+
+				testEvaluationWithOutputInHeader(
+						image, "{\"labels\":[{\"giant panda\":0.98649305}]}");
```

Comment:
All tests fail for me. For example this one:
```
Expected :{\"labels\":[{\"giant panda\":0.98649305}]}
Actual   :{\"labels\":[{\"giant panda\":0.9864959}]}
```
However I won't mind that it is something related to my Windows OS.
But at the same time I would like to understand "why?" and make our app cross-OS.
Would you mind pointing where to look for this particular problem?

Or there has been something changes in the probability calculation and now Tensorflow knows better that our panda is really sad 😄 

Nevertheless I still think we have to `@Ignore` these tests as well. They need Internet connection and they always download 52 MB of model to train:
```
2017-09-19 17:19:16.664  INFO 7140 --- [           main] s.a.t.p.TensorflowProcessorConfiguration : Loading TensorFlow graph model: URL [http://dl.bintray.com/big-data/generic/tensorflow_inception_graph.pb]
2017-09-19 17:19:19.698  INFO 7140 --- [           main] p.ImageRecognitionProcessorConfiguration : Load ImageRecognitionTensorflowInputConverter
```